torch/nn/parallel/distributed.py

```diff
@@ -600,16 +608,13 @@ def __init__(
         self.static_graph = False
         self.dim = dim
         self.module = module
-        self.device = list(self.module.parameters())[0].device
+        self.device = list(self.module_parameters)[0].device
```

Comment:
No need for `list` when `self.module_parameters` is already a `list`.

Also, what happens if you ignore all of the module's parameters? We should handle that case gracefully instead of `self.device = ...` causing an index out of bounds error.

Reply:
Added a

```
        if not self._module_parameters:
            self._log_and_throw(
                RuntimeError,
                "DistributedDataParallel is not needed when a module doesn't have"
                "any parameters that are not ignored."
            )

```
at the beginning,